resources/centos.rb

```diff
@@ -78,6 +78,29 @@
   # Include 'yum', and 'yum-centos' recipies
   # 'yum' will apply our changes to the main config file
   # 'yum-centos' install the remaining repositories and apply our configuration
-  include_recipe 'yum'
-  include_recipe 'yum-centos'
+  if repo_resource_exist?('base')
+    edit_resource!(:yum_globalconfig, '/etc/yum.conf') do
+      node['yum']['main'].each do |config, value|
+        send(config.to_sym, value)
+      end
+    end
+
+    node['yum-centos']['repos'].each do |repo|
+      next unless node['yum'][repo]['managed']
+      edit_resource!(:yum_repository, repo) do
+        node['yum'][repo].each do |config, value|
+          case config
+          when 'managed' # rubocop: disable Lint/EmptyWhen
```

Comment:
Is this empty `when` just an explicit "don't do anything here"?

Reply:
I think so, I just copied from the yum-centos cookbook and I know if its not there it won't work.

Reply:
Ah, the `else` clause does other stuff. Gotcha.